pkg/local-gadget-manager/local-gadget-manager.go

```diff
@@ -0,0 +1,309 @@
+	if !ok {
+		return fmt.Errorf("unknown gadget")
+	}
+	_, ok = l.tracers[name]
+	if ok {
+		fmt.Printf("Error: trace already exists")
+		return nil
+	}
+
+	traceResource := &gadgetv1alpha1.Trace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: "gadget",
+		},
+		Spec: gadgetv1alpha1.TraceSpec{
+			Node:   "local",
+			Gadget: gadget,
+			Filter: &gadgetv1alpha1.ContainerFilter{
+				Namespace:     "local",
+				Podname:       containerFilter,
+				ContainerName: "",
+				Labels:        map[string]string{},
+			},
+			RunMode:    "Manual",
+			OutputMode: "Stream",
+		},
+	}
+
+	l.tracers[name] = tracer{
+		gadget:        gadget,
+		name:          name,
+		factory:       factory,
+		traceResource: traceResource,
+		gadgetStream:  stream.NewGadgetStream(),
+	}
+	return nil
+}
+
+func (l *LocalGadgetManager) Operation(name, opname string) error {
+	tracer, ok := l.tracers[name]
+	if !ok {
+		return fmt.Errorf("unknown trace")
+	}
+
+	if opname != "" {
+		gadgetOperation, ok := tracer.factory.Operations()[opname]
+		if !ok {
+			return fmt.Errorf("Unknown operation %q", opname)
+		}
+		tracerNamespacedName := tracer.traceResource.ObjectMeta.Namespace +
+			"/" + tracer.traceResource.ObjectMeta.Name
+		gadgetOperation.Operation(tracerNamespacedName, tracer.traceResource)
+	}
+
+	fmt.Printf("State: %s\n", tracer.traceResource.Status.State)
+	fmt.Printf("OperationError: %s\n", tracer.traceResource.Status.OperationError)
+	fmt.Printf("Output: %s\n", tracer.traceResource.Status.Output)
+
+	return nil
+}
+
+func (l *LocalGadgetManager) Show(name string) error {
+	return l.Operation(name, "")
+}
+
+func (l *LocalGadgetManager) Delete(name string) error {
+	tracer, ok := l.tracers[name]
+	if !ok {
+		return fmt.Errorf("unknown trace")
+	}
+
+	factory, ok := l.traceFactories[tracer.gadget]
+	if !ok {
+		return fmt.Errorf("unknown gadget")
+	}
+	factory.Delete("gadget/" + name)
+	delete(l.tracers, name)
+	return nil
+}
+
+func (l *LocalGadgetManager) PublishEvent(tracerID string, line string) error {
+	name := strings.TrimPrefix(tracerID, "trace_gadget_")
+	t, ok := l.tracers[name]
+	if !ok {
+		return fmt.Errorf("cannot find tracer: unknown tracer %q", tracerID)
+	}
+
+	t.gadgetStream.Publish(line)
+	return nil
+}
+
+func (l *LocalGadgetManager) Stream(name string, stop chan struct{}) (chan string, error) {
+	t, ok := l.tracers[name]
+	if !ok {
+		return nil, fmt.Errorf("cannot find trace %q", name)
+	}
+
+	out := make(chan string)
+
+	ch := t.gadgetStream.Subscribe()
+
+	go func() {
+		if stop == nil {
+			for len(ch) > 0 {
+				line := <-ch
+				out <- line.Line
+			}
+			t.gadgetStream.Unsubscribe(ch)
+			close(out)
+		} else {
+			for {
+				select {
+				case <-stop:
+					t.gadgetStream.Unsubscribe(ch)
+					close(out)
+					return
+				case line := <-ch:
+					out <- line.Line
+				}
+			}
+		}
+	}()
+	return out, nil
+}
+
+func (l *LocalGadgetManager) AddContainer(notif runcfanotify.ContainerEvent) {
```

Comment:
LocalGadgetManager already has `ContainerAdd`, I think these two can be easily confused. What about calling it `AddContainerRuncNotif`?

Reply:
Ok. And I will unexport the function too (`addContainerRuncNotif` and `RemoveContainerRuncNotif`) since they are not used outside of the package.